src/js/components/agency/overview/AgencyOverview.jsx

```diff
@@ -23,6 +30,44 @@ export default class AgencyOverview extends React.Component {
             logo = (<img src={this.props.agency.logo} alt={this.props.agency.name} />);
         }
 
+        // Move props to variables for readability
+        const budgetAuthority = this.props.agency.budgetAuthority;
+        const federalBudget = this.props.agency.federalBudget;
+        const fy = parseInt(this.props.agency.activeFY, 10);
+        const quarter = parseInt(this.props.agency.activeFQ, 10);
+
+        // Generate "as of" date
+        const endOfQuarter = convertQuarterToDate(quarter, fy);
+        const asOfDate = moment(endOfQuarter, "YYYY-MM-DD").format("MMMM D, YYYY");
+
+        // Generate Budget Authority string
+        const budgetAuthorityAmount = MoneyFormatter.calculateUnitForSingleValue(budgetAuthority);
+        const formattedBudgetAuthority = `${MoneyFormatter.formatMoneyWithPrecision(
+            budgetAuthority / budgetAuthorityAmount.unit, 1)}
+            ${capitalize(budgetAuthorityAmount.longLabel)}`;
+
+        // Generate Percentage string
+        const percentage = MoneyFormatter.calculateTreemapPercentage(
+            budgetAuthority, federalBudget);
+        const percentageElement = (
+            <span className="authority-statement-percentage">{percentage}</span>
+        );
+
+        // Generate visualization parameters
+        let visualizationWidth = 0;
+        if (this.containerDiv) {
+            visualizationWidth = this.containerDiv.getBoundingClientRect().width;
```

Comment:
This measures the div of the last render cycle - this currently works because of a bug in the sticky sidebar that triggers a re-render on every scroll event, so the measurement occurs extremely frequently. The scroll bug is fixed in #329 which means this measurement occurs on mount (but does nothing because the div doesn't exist yet) and after the API call is parsed (which gives it a good initial reading), but won't update again on browser resize.

A better way is to measure after `componentDidMount` (which is triggered after the initial `render`), then attach a window resize event to measure on resize. Store the width in state and use that value.